kafka-rest/src/main/java/io/confluent/kafkarest/Utils.java

```diff
@@ -43,4 +53,44 @@ public static int errorCodeFromProducerException(Throwable e) {
     }
   }
 
+  public static RestException convertProducerException(KafkaException exception) {
+    if (StringUtil.startsWithIgnoreCase(exception.getMessage(), "Invalid partition")) {
+      throw Errors.partitionNotFoundException();
+    } else if (exception instanceof AuthenticationException) {
+      return Errors.authenticationException(exception.getMessage());
```

Comment:
Maybe it wouldn't hurt to reduce the code duplication in between these three methods and have a single method to check instances and return `Errors.authenticationException`, `Errors.authorizationException` or `Errors.kafkaErrorException`.